crates/red_knot_python_semantic/src/types/display.rs

```diff
@@ -247,6 +247,31 @@ impl std::fmt::Debug for DisplayIntersectionType<'_> {
     }
 }
 
+impl<'s, 'db> TypeList<'db> {
+    pub(crate) fn display(&'s self, db: &'db dyn Db) -> DisplayConflictingTypes<'s, 'db> {
+        DisplayConflictingTypes { types: self, db }
+    }
+}
+
+pub(crate) struct DisplayConflictingTypes<'l, 'db> {
+    types: &'l TypeList<'db>,
+    db: &'db dyn Db,
+}
+
+impl std::fmt::Display for DisplayConflictingTypes<'_, '_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut iter = self.types.0.iter();
+        if let Some(first) = iter.next() {
+            write!(f, "'{}'", first.display(self.db))?;
+            for other in iter {
+                f.write_str(", ")?;
+                write!(f, "'{}'", other.display(self.db))?;
+            }
+        }
+        Ok(())
```

Comment:
I'm not sure how many more times we should write this before introducing a helper similar to `f.debug_list`.

* Create a `FormatterJoinExtension` trait with a single method `join(&mut self, separator: &str) -> Join`
* Create a `Join` struct with:
  * `entry(&mut self, item: &dyn std::fmt::Display)`
  * `entries<I, F>(&mut self, items: I) where I: IntoIterator<Item = F>, F: std::fmt::Display`
  * `finish(self) -> Result<()>`

Reply:
I added this, and converted everywhere we had the "join" pattern in `types/display.rs` to use it.